framework/include/utils/PiecewiseByBlockLambdaFunctor.h

```diff
@@ -58,6 +58,9 @@ class PiecewiseByBlockLambdaFunctor : public Moose::FunctorBase<T>
 
   bool hasBlocks(SubdomainID id) const override;
 
+  bool supportsFaceArg() const override final { return true; }
+  bool supportsElemSideQpArg() const override final { return true; }
```

Comment:
same

Reply:
poll all of them? it s a map right?

Reply:
It has a map of block ID to a `std::function` for each arg type, so it doesn't actually have other `FunctorBase`s.

Reply:
umm so is std::function always good then?

Reply:
No it's not, but we have no idea what the users will put in their functor materials. Someone could put something that throws an error for one of the argument types. I think we just won't be able to intercept these with `supports...`. @lindsayad?

Reply:
I think so too. default true is fine

Reply:
yea, we don't know what will happen in the arbitrary lambda the users are providing